src/box/lua/config/applier/box_cfg.lua

```diff
@@ -273,6 +259,22 @@ local function set_ro_rw(config, box_cfg)
             assert(false)
         end
     elseif failover == 'supervised' then
+        -- If replication.bootstrap_strategy = supervised or
+        -- native, all the leader management actions are performed
+        -- by the failover coordinator (an external agent).
+        --
+        -- Don't do anything in the case.
+        local bootstrap_strategy = configdata:get(
+            'replication.bootstrap_strategy', {use_default = true})
+        local no_op_strategies = {
+            supervised = true,
+            native = true,
+        }
+        if no_op_strategies[bootstrap_strategy] then
+            box_cfg.read_only = true
+            return
+        end
```

Comment:
I'd like the user to perform less configuration.
Thus, I want the `box_cfg` applier automatically perform `box.internal.runtime_priv.grant('guest', 'execute', 'lua_call', 'failover.execute')` when working with `replication.bootstrap_strategy: supervised/native` and `replication.failover: supervised`. At first, I'd like to know your opinion on this to reduce the amount of configuration needed. 

Also, this seems like a perfect place for such logic, but `no_op_strategies` name seems kinda strange for any actions other than `box_cfg.read_only = true`. Thus, I suggest changing the name somehow.

```suggestion
        local supervised_strategies = {
            supervised = true,
            native = true,
        }
        if supervised_strategies[bootstrap_strategy] then
            box_cfg.read_only = true
            return
        end
```

Reply:
Even if it works to some extent, I'm against of calling `box.internal.runtime_priv.grant()` from anywhere outside of the `runtime_priv` configuration applier. They reason is that we do the following in the applier:

* collect *all* the needed runtime privileges
* reset the runtime privileges
* set the new privileges

This way we don't have to calculate a difference between actual and target privileges. It simplifies the implementation of the applier and *also* simplifies the underlying API (this way we have just two functions `reset` and `grant`, where `reset` just drops everything).

However, it adds a little constraint: we have to assign all the privileges in one pass or, at least, within one configuration applying iteration. But it is much more clear to keep everything within the relevant applier.

Reply:
Renamed the strategies list. Also, found that we shouldn't go to RO on reconfiguration, fixed, covered with a test.

```diff
diff --git a/src/box/lua/config/applier/box_cfg.lua b/src/box/lua/config/applier/box_cfg.lua
index ce7eabd8a5..d8ec06666a 100644
--- a/src/box/lua/config/applier/box_cfg.lua
+++ b/src/box/lua/config/applier/box_cfg.lua
@@ -268,12 +268,19 @@ local function set_ro_rw(config, box_cfg)
         -- an RW instance.
         local bootstrap_strategy = configdata:get(
             'replication.bootstrap_strategy', {use_default = true})
-        local startup_in_ro_strategies = {
+        local externally_managed_mode_strategies = {
             supervised = true,
             native = true,
         }
-        if startup_in_ro_strategies[bootstrap_strategy] then
-            box_cfg.read_only = true
+        if externally_managed_mode_strategies[bootstrap_strategy] then
+            if is_startup then
+                box_cfg.read_only = true
+            else
+                -- Don't change the read_only flag on the
+                -- reconfiguration. It is managed by an external
+                -- agent.
+                box_cfg.read_only = nil
+            end
             return
         end
 
diff --git a/test/config-luatest/native_bootstrap_strategy_test.lua b/test/config-luatest/native_bootstrap_strategy_test.lua
index 544ba703b5..0a2566686f 100644
--- a/test/config-luatest/native_bootstrap_strategy_test.lua
+++ b/test/config-luatest/native_bootstrap_strategy_test.lua
@@ -417,6 +417,14 @@ g_supervised_and_native.test_failover_supervised = function(g)
     end)
     verify_cluster_mode(cluster, {'rw', 'ro', 'ro'})
     verify_bootstrap_leader(cluster, 'i-001')
+
+    -- Reconfiguration doesn't drop the RW mode to RO.
+    cluster['i-001']:exec(function()
+        local config = require('config')
+
+        config:reload()
+        t.assert_equals(box.info.ro, false)
+    end)
 end
 
 -- Same as the previous test case, but cover a replicaset with one
@@ -469,4 +477,12 @@ g_supervised_and_native.test_failover_supervised_singleton = function(g)
     end)
     verify_cluster_mode(cluster, {'rw'})
     verify_bootstrap_leader(cluster, 'i-001')
+
+    -- Reconfiguration doesn't drop the RW mode to RO.
+    cluster['i-001']:exec(function()
+        local config = require('config')
+
+        config:reload()
+        t.assert_equals(box.info.ro, false)
+    end)
 end
```

Reply:
Thanks. Closing and the related logic will be added in #11381